app/controllers/impressionist_controller.rb

```diff
@@ -154,7 +154,7 @@ def session_hash
         id = request.session_options[:id]
       end
 
-      unless id.is_a? String
+      if id.respond_to?(:cookie_value)
         id = id.cookie_value if Rack::Session::SessionId.const_defined?(:ID_VERSION) && Rack::Session::SessionId::ID_VERSION == 2
```

Comment:
I think this is a little simpler

```suggestion
        id = id.cookie_value
```